node/nodejs_intro/README.md

```diff
@@ -1,65 +1,98 @@
+</details>
+<p>&nbsp;</p>
 
-These languages are C, Fortran, PASCAL and C++ to name a few. These languages are directly "compiled" into the Binary machine code our computers understand.
+In the diagram below (you can ignore 'Assembly Lanugage' for now):
+- 💻 `Hardware` is your computer
+- 0️⃣1️⃣ `Machine Code` is the binary code that is giving your computer instructions
+- 📝 `High-Level Lanugages` are sitting on top of everything, getting *compiled* into binary code for the computer to understand.
 
-Your operating system is basically a computer program that is written in one of these languages.
 
 ![Machine Code](assets/machinecode.png)
 
-### Interpreted Languages
+`Check For Understanding:` Why do we have higher level programming languages?
+<details>
+<summary>
+    Answer
+  </summary>
+These languages allow us to "abstract" away the binary code that underlies all of our computer programs, so we can write code in a more human readable way! Who has time to write 0s and 1s all day?!
+</details>
+<p>&nbsp;</p>
+
+## Interpreted Languages
+
+Now, where does Javascript fit into all this? Well, ✨ Javascript is special ✨. Javascript is also a high level language, in that it's human-readable and it automates a lot of things like memory allocation, but its code does **not** directly compile to binary. It is referred to as an `"interpreted"` language, in other words, you need an interpreter to understand how to run it!
+
+Javascript was developed in the 1990s to add interactivity, animations, and little bits of automation to websites.
 
-Now, where does Javascript fit into all this? Does our Javascript code compile to binary? Unfortunately, it does not.
+*Before Javascript, remember websites looking like this??*
+![Old Website](assets/oldwebsite.jpeg)
+*Altavista website by Christiaan Colen - via [link](https://www.flickr.com/photos/christiaancolen/18598795371)*
 
-Javascript was developed in the 1990s to add interactions to websites. Your Web browser is just a computer program. It had to be written by someone using a High Level Programming Language.
+But, Javascript was first built *just* for use in Web browsers, it didn't need to compile to binary because it was not going to be run on a machine - it would be interpreted by and run in the Web browser.
 
-It's a special type of program that does the following:
 
-1. Make network requests to the Internet and retrieve data.
-2. Interpret and display the data it recieves.
+## Web Browser
+Your Web browser (Google Chrome, Safari, Firefox, etc.) is just a ✨ special ✨ computer program (also written in a High-Level programming language), that does the following:
+
+1. `Make network requests to the Internet and retrieve data.`
+2. `Interpret and display the data it recieves.`
 
 **Javascript, HTML and CSS are all *interpreted* by your web browser on the fly.**
 
+![Javascript in Browser](assets/javascriptbrowser.png)
+
 Javascript ONLY works if a computer program such as a Web browser interprets it.
```

Comment:
Is it possible to have a check for understanding or short group activity here before we move to objective 2?

Reply:
👍